src/APICaller.js

```diff
@@ -0,0 +1,21 @@
+async function fetchWeatherData(url) {
```

Comment:
This comment is related to file name -> keep consistent with file names. Usually in javascript you would call files like apiCaller, rather than APICaller. But I would also suggest calling this service just as an api.js, which would implement fetchWeatherData() {}